static/js/xmleditor/xsd2json.js

```diff
@@ -466,25 +480,77 @@ SchemaManager.prototype.resolveSchema = function(schema, name) {
 	return this;
 };
 
-SchemaManager.prototype.mergeType = function(base, type) {
+SchemaManager.prototype.mergeType = function(base, type, mergeMode) {
 	for (var key in type) {
 		if (type.hasOwnProperty(key)) {
 			var value = type[key];
 			if (value != null && base[key] == null){
 				base[key] = value;
-			} else if ($.isArray(value) && $.isArray(type[key])){
-				base[key] = base[key].concat(value);
+			} else if ($.isArray(value) && $.isArray(base[key])){
+				base[key] = this.mergeTypeArray(base[key], value, key, mergeMode);
 			}
 		}
 	}
 };
 
+SchemaManager.prototype.mergeTypeArray = function(baseArray, typeArray, key, mergeMode) {
+	var mergeFunction = this['mergeType_' + key + '_' + mergeMode];
+	if (mergeFunction === undefined) {
+		throw Error('Invalid definition key ' + JSON.stringify(key) + ' or merge mode ' + JSON.stringify(mergeMode));
+	}
+	return mergeFunction.call(this, baseArray, typeArray);
+};
+
+SchemaManager.prototype.mergeType_choices_extension = function(baseArray, typeArray) {
+	return baseArray.concat(typeArray);
+};
+
+SchemaManager.prototype.mergeType_choices_restriction = function(baseArray, typeArray) {
+	return baseArray.concat(typeArray);
+};
+
+SchemaManager.prototype.mergeType_attributes_extension = function(baseArray, typeArray) {
+	return baseArray.concat(typeArray);
+};
+
+SchemaManager.prototype.mergeType_attributes_restriction = function(baseArray, typeArray) {
+	var baseNames = {};
+	for (var i = 0; i < baseArray.length; i++) {
+		baseNames[baseArray[i].name] = true;
+	}
+	var typeResult = [];
+	for (var j = 0; j < typeArray.length; j++) {
+		var typeItem = typeArray[j];
+		if (!baseNames[typeItem.name]) {
+			typeResult.push(typeItem);
+		}
+	}
+	return typeResult.concat(baseArray);
+};
+
```

Comment:
My understanding is that the behavior can vary based component type, but the difference is only implemented for 2 of them right now